concepts/pointers/about.md

```diff
@@ -1,3 +1,155 @@
+
+```go
+a = 3
+```
+
+The piece of memory that `a` is associated with now will have the value `3`.
+
+## Pointers
+
+While variables allow us to refer to values in memory, sometimes it's usefull to know the memory address of the value that the variable refers. Pointers hold the memory addresses of those values. A pointer declaration looks like this:
+
+
+```go
+var p *int// 'p' can hold the memory address of an integer
+```
+
+Here we declaring a variable `p` of type `*int`. The type `*int` means "pointer to int". In other words, we are saying that `p` is a pointer to int, which means that `p` will hold the memory address of an integer. The zero value of pointers is `nil`. A `nil` pointer holds no memory address.
+
+### Getting a pointer to a variable
+
+The operator `&` gives the memory address of its operand. This means that to get the memory address of the value of a variable `a`, we can do `&a`. `&a` will create a pointer to `a`:
+
+```go
+var a int = 2
+var p *int = &a // the variable 'p' wil hold the memory address of 'a'
+```
+
+We could also have used the sorthand syntax for the variables. The following code is equivalent to the previous code:
+
+```go
+a := 2
+p := &a // the variable 'p' wil hold the memory address of 'a'
+```
+
+The operation `&a` is sometimes called "referencing", as we are getting a reference (pointer) to the value of a variable.
+
+### Accessing the value via a pointer
+
+When we have a pointer, we might want to know the value that is in the memory address of the pointer. We can do this using the `*` operator:
+
+```go
+var a int = 2
+var p *int = &a // the variable 'p' wil hold the memory address of 'a'
+
+var b int = *p // b == 2
+```
+
+Or using the shorthand syntax for the variables:
+
+```go
+a := 2
+p := &a // the variable 'p' wil hold the memory address of 'a'
+
+b := *p // b == 2
+```
+
+The operation `*p` fetches the value stored at the memory address stored in `p`. This operation is often called "dereferencing".
+
+We can also use the derefering operator to assign a new value to the memory address referenced by the pointer:
+
+
+```go
+
+a  := 2          // declare int variable 'a' and assign it the value of 2 
+pa := &a         // 'pa' now refers to the memory address of 'a'  
+*pa = *pa + 2    // increment by 2 the value at memory address 'pa'
+
+fmt.Println(a)   // Output: 4
+                 // 'a' will have the new value that was changed via the pointer!
+```
+
+Assigning to `*pa` will change the value stored at the memory address `pa` holds. Since `pa` holds the memory address of `a`, by assigning to `*pa` we are effectively changing the value of `a`! 
+
+
+A note of caution however: always check if a pointer is not `nil` before dereferencing. Dereferencing a `nil` pointer will make the program panic at runtime!
+
+### Pointers to structs
+
+So far we've only seen pointers to primitive values. We can also create pointers for structs:
+
+```go
+type Point struct {
```

Comment:
I'd personally recommend using another name for this struct type as the `Point` name is too similar to the pointer concept and may confuse readers. Perhaps a `Person` struct instead?

Reply:
That's a good point. After writing it down I also thought about that. I'll change it :)

Reply:
Sorry to butt in! I was just poking around at the open pull requests. Appreciate your work here!